ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -667,3 +702,110 @@ wakeupDecision st _now =
     decisionState = st,
     decisionJobs  = []
   }
+
+
+-- | Classify if a node is in promiscuous mode.
+--
+-- A node is not in promiscuous mode only if: it doesn't use ledger peers, peer
+-- sharing, the set of bootstrap peers is empty.
+--
+readAssociationMode
+  :: MonadSTM m
+  => STM m UseLedgerPeers
+  -> PeerSharing
+  -> UseBootstrapPeers
+  -> PeerSelectionSetsWithSizes peeraddr
+  -> STM m AssociationMode
+readAssociationMode
+  readUseLedgerPeers
+  peerSharing
+  useBootstrapPeers
+  PeerSelectionView {
+    viewKnownBootstrapPeers = (knownBootstrapPeers, _)
+  }
+  =
+  do useLedgerPeers <- readUseLedgerPeers
+     pure $
+       case (useLedgerPeers, peerSharing, useBootstrapPeers) of
+         (DontUseLedgerPeers, PeerSharingDisabled, DontUseBootstrapPeers)
+           -> LocalRootsOnly
+         (DontUseLedgerPeers, PeerSharingDisabled, UseBootstrapPeers {})
+           |  Set.null knownBootstrapPeers
+           -> LocalRootsOnly
+         _ -> Unrestricted
+
+
+outboundConnectionsState
+    :: Ord peeraddr
+    => AssociationMode
+    -> PeerSelectionSetsWithSizes peeraddr
+    -> PeerSelectionState peeraddr peerconn
+    -> OutboundConnectionsState
+outboundConnectionsState
+    associationMode
+    PeerSelectionView {
+        viewEstablishedPeers = (viewEstablishedPeers, _),
+        viewEstablishedBootstrapPeers = (viewEstablishedBootstrapPeers, _),
+        viewActiveBootstrapPeers = (viewActiveBootstrapPeers, _)
+      }
+    PeerSelectionState { localRootPeers,
+                         publicRootPeers,
+                         bootstrapPeersFlag
+                       }
+    =
+    case (associationMode, bootstrapPeersFlag) of
+      {-
+       -- genesis mode
+       -- TODO: issue #4846
+      (LocalRootsOnly, _)
+                |  numberOfActiveBigLedgerPeers >= targetNumberOfActiveBigLedgerPeers
+                -> TrustedStateWithExternalPeers
+
+                |  otherwisej
+                -> UntrustedState
+      -}
+
+      (LocalRootsOnly, _)
+        |  -- we are only connected to trusted local root
+           -- peers or bootstrap peers
+           viewEstablishedPeers `Set.isSubsetOf` (viewEstablishedBootstrapPeers <> trustableLocalRootSet)
```

Comment:
If we are configured to use `LocalRootsOnly` we shouldn't have any established bootstrap peers so they shouldn't be included in the check.